packages/core/src/util.ts

```diff
@@ -1,18 +1,43 @@
 import type { EventPlugin } from './plugin';
 import type { Timeline } from './timeline';
 
-export const chunk = (array: any[], size: number) => {
-  if (!array.length || !size) {
+const sizeOf = (obj: unknown): number => {
+  const size = encodeURI(JSON.stringify(obj)).split(/%..|./).length - 1;
+  return size / 1024;
+};
+
+export const chunk = (array: any[], count: number, maxKB?: number) => {
+  if (!array.length || !count) {
     return [];
   }
 
-  let index = 0;
-  let resIndex = 0;
-  const result = new Array(Math.ceil(array.length / size));
+  let currentChunk = 0;
+  let rollingKBSize = 0;
+  const result: any[] = array.reduce(
+    (chunks: any[][], item: any, index: number) => {
+      if (maxKB !== undefined) {
+        rollingKBSize += sizeOf(item);
```

Comment:
wont this be more accurate to do in plain bytes rather than kb? we wont have to do the `/ 1024` everytime we add bytes that way